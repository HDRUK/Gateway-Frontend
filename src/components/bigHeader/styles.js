import styled from "styled-components";
import { HeaderWrapper } from "../../styles/carbonComponents.js";

export const BigHeaderWrapper = styled(HeaderWrapper)`
    display: block;
    height: 33.5rem;
    padding-top: 2rem;
`;

export const BigHeaderImage = styled.img`
    height: 6.4375rem;
    display: block;
`;
