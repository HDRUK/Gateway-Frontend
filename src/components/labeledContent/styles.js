import styled from "styled-components";

export const LabeledContentWrapper = styled.div`
    display: inline-block;

    h2,
    p {
        padding-top: 0.125rem;
        padding-right: 0.5rem;
    }
`;

export const HeaderLabel = styled.h2`
    font-size: 0.625rem;
    line-height: 0.75rem;
    color: #979797;
    text-transform: uppercase;
`;

export const ContentText = styled.p`
    font-size: 0.875rem;
    line-height: 1.0625rem;
    color: #0f62fe;

    ${p => !p.lowercase && "text-transform: uppercase"}
`;
