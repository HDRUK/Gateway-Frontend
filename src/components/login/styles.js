import styled from "styled-components";

export const LoginWrapper = styled.span`
    padding: 0 2rem;
    text-align: center;
`;
