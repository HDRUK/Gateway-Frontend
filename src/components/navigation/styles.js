import styled from "styled-components";
import { HeaderNavigation } from "carbon-components-react";

export const NavigationWrapper = styled.div`
    width: 100%;
    height: ${p => p.theme.sizes.navigation}rem;
    border-bottom: 0.0625rem solid ${p => p.theme.colors.border};
`;

export const NavigationHeader = styled(HeaderNavigation)`
    :before {
        content: unset;
    }

    @media (max-width: 66rem) {
        display: block;
    }
`;

export const NavigationItem = styled.div`
    height: 100%;
    min-height: 2.875rem;
    padding: 1rem;
    line-height: 0.875rem;
    text-decoration: none;
    color: ${p => p.theme.text.primary};

    ${p => p.active && `border-bottom: 0.25rem solid ${p.theme.colors.primarySolid}`};
    font-weight: ${p => p.active && p.theme.text.bold};

    :hover,
    :active {
        border-bottom: 0.25rem solid ${p => p.theme.colors.primarySolid};
        font-weight: ${p => p.theme.text.bold};
    }
`;
