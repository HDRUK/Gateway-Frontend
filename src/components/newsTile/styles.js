import styled from "styled-components";

export const NewsTileWrapper = styled.div`
    display: inline-block;
    width: 26rem;
    height: 27.5rem;
    overflow: hidden;
    border-radius: 0.125rem;
    box-shadow: 0 0 0.25rem 0 ${p => p.theme.colors.shadow};
`;
