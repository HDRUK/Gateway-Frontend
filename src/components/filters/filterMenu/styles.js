import styled from "styled-components";
import { AccordionBlock } from "../../../styles/carbonComponents";

export const ExpandingAccordionBlock = styled(AccordionBlock).attrs(p => ({
    style: {
        paddingBottom: p.expandheight ? `${p.expandheight}px` : "1rem"
    }
}))`
    display: inline-block;
    padding-bottom: "1rem";
`;
