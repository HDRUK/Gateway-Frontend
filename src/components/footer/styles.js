import styled from "styled-components";
import { WidthWrapper, Logo } from "../../styles/styles.js";

export const FooterWrapper = styled.div`
    height: ${p => p.theme.sizes.footer}rem;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    color: white;
    padding-top: 4rem;
    padding-bottom: 1rem;
    font-size: 1.1rem;
    position: relative;
`;

export const FooterContent = styled(WidthWrapper)`
    display: block;
    margin: 0 auto 0;
    max-width: 62rem;
    padding: 0 2rem;
`;

export const FooterBlock = styled.div`
    display: inline-block;
    box-sizing: border-box;
    width: 33.33%;
    height: 100%;
    vertical-align: top;
    padding: 0 4rem 0 0;
    line-height: 1.1875rem;

    :last-of-type {
        padding-right: 0;
        padding-left: 4rem;
    }
`;

export const FooterText = styled.div`
    width: 100%;
    margin: 0 0 0.4rem 0;
`;

export const FooterImage = styled(Logo)`
    max-width: 8rem;
    max-height: 5rem;
`;
