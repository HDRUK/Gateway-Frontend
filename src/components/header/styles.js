import styled from "styled-components";
import { LinkNoDecoration } from "../../styles/styles.js";

export const HeaderAlignment = styled.div`
    width: 100%;
`;

export const HeaderAlignmentItem = styled.div`
    height: 100%;
    float: ${p => (p.right ? "right" : "left")};
`;

export const HeaderSpacer = styled.div`
    height: 100%;
    width: 6rem;
`;

export const HeaderLink = styled.a`
    :hover {
        text-decoration: underline;
        text-decoration-color: ${p => p.theme.colors.white};
    }
`;
