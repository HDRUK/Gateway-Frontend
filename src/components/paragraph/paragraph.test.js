import React from "react";
import { create } from "react-test-renderer";
import Paragraph from "./paragraph";
import { ParagraphText, DarkText } from "../../styles/styles.js";

describe("<Paragraph> ", () => {
    let renderedComponent;

    beforeEach(() => {
        renderedComponent = create(<Paragraph />);
    });

    it("should test the NewsTile Snapshot", () => {
        expect(renderedComponent).toMatchSnapshot();
    });

    it("should check the correct style elements are displayed", () => {
        const paragraphText = renderedComponent.root.findAllByType(ParagraphText);
        const darkText = renderedComponent.root.findAllByType(DarkText);

        expect(paragraphText).toHaveLength(1);
        expect(darkText).toHaveLength(1);
    });
});
