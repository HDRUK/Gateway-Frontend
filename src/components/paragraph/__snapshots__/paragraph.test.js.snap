// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph>  should test the NewsTile Snapshot 1`] = `
<text
  className="sc-EHOje cUVptH"
>
  <p
    className="sc-ifAKCX hSPtPb"
  />
</text>
`;
